demo-app/src/tx_hooks_impl.rs

```diff
@@ -1,24 +1,25 @@
-use crate::tx_verifier::{Transaction, VerifiedTx};
+use sov_app_template::{TxHooks, VerifiedTx};
 use sov_modules_api::{Context, Spec};
 use sov_state::WorkingSet;
 
-/// TxHooks allows injecting custom logic into a transaction processing pipeline.
-pub(crate) trait TxHooks {
-    type Context: Context;
+use crate::tx_verifier_impl::Transaction;
 
-    /// pre_dispatch_tx_hook runs just before a transaction is dispatched to an appropriate module.
-    fn pre_dispatch_tx_hook(
-        &mut self,
-        tx: Transaction<Self::Context>,
-        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
-    ) -> anyhow::Result<VerifiedTx<Self::Context>>;
+pub(crate) struct AppVerifiedTx<C: Context> {
+    pub(crate) pub_key: C::PublicKey,
+    pub(crate) sender: C::Address,
+    pub(crate) runtime_msg: Vec<u8>,
+}
 
-    /// post_dispatch_tx_hook runs after the tx is dispatched to an appropriate module.
-    fn post_dispatch_tx_hook(
-        &mut self,
-        tx: VerifiedTx<Self::Context>,
-        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
-    );
+impl<C: Context> VerifiedTx for AppVerifiedTx<C> {
```

Comment:
(non-blocking): This trait looks very similar to the `BlobWithSender` trait. Maybe in the future we can unify the two of them?

Reply:
I will have a look!